compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -120,16 +135,48 @@ object Completion {
       }
 
     val describedCompletions = describeCompletions(completions)
+    val backtickedCompletions = describedCompletions.map(backtickCompletions)
+
     val offset = completionOffset(path)
 
     interactiv.println(i"""completion with pos     = $pos,
                           |                prefix  = ${completer.prefix},
                           |                term    = ${completer.mode.is(Mode.Term)},
                           |                type    = ${completer.mode.is(Mode.Type)}
-                          |                results = $describedCompletions%, %""")
-    (offset, describedCompletions)
+                          |                results = $backtickCompletions%, %""")
+    (offset, backtickedCompletions)
   }
 
+  def backtickCompletions(completion: Completion) =
+    if needsBacktick(completion.label) then
+      completion.copy(label = s"`${completion.label}`")
+    else
+      completion
+
+  // This borrows from Metals, which itself borrows from Ammonite. This uses
+  // the same apprach, but some of the utils that already exist in Dotty.
+  // https://github.com/scalameta/metals/blob/main/mtags/src/main/scala/scala/meta/internal/mtags/KeywordWrapper.scala
+  // https://github.com/com-lihaoyi/Ammonite/blob/73a874173cd337f953a3edc9fb8cb96556638fdd/amm/util/src/main/scala/ammonite/util/Model.scala
+  private def needsBacktick(s: String) =
+    val chunks = s.split("_", -1)
+    
+    val validChunks = chunks.zipWithIndex.forall { case (chunk, index) =>
+      chunk.forall(Chars.isIdentifierPart) ||
+      (chunk.forall(Chars.isOperatorPart) &&
+        index == chunks.length - 1 &&
+        !(chunks.lift(index - 1).contains("") && index - 1 == 0))
+    }
+    
+    val validStart =
+      Chars.isIdentifierStart(s(0)) || chunks(0).forall(Chars.isOperatorPart)
+
+    val valid = validChunks && validStart && !keywords.contains(s)
+
+    !valid
+  end needsBacktick
+
+  private lazy val keywords = Tokens.keywords.map(Tokens.tokenString)
```

Comment:
I was also a bit unsure about this. Is there any better way to do a token lookup with a string?

Reply:
You could also try `dotty.tools.dotc.core.StdNames.nme.keywords.map(_.toString)`